pkg/asset/installconfig/powervs/regions.go

```diff
@@ -0,0 +1,86 @@
+package powervs 
+
+import (
+	"github.com/powervs/powervs-sdk-go/powervs/endpoints"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/openshift/installer/pkg/rhcos"
+)
+
+// TODO(cklokman): aws uses another repo to provide endpoints, which seems to be used to query
+//		   region information.  For my testing I implemented a very simple version of this
+//		   that returns hard coded data to satisfy this, and return similar data for powervs
+//		   I am uncertain the direction to take here, but don't believe hardcoding that data
+//		   in this file is the correct solution.  It may be benificial to hold another repo
+//		   in the same way as aws, or make api calls here to grab the required information.
+//
+
+// TODO(cklokman): This section came from aws and returns a map of regions, which will need to be
+//                 a subset of powervs regions and the regions where RHEL CoreOS images are available.
+//                 This process will be different from AWS, because of the method IBM is using for the
+//                 CoreOS images.  It may be the case that all regions will have CoreOS images present
+//                 in that case, this function becomes quite straight forward.  For the moment I am
+//                 leaving this logic here, because this logic may be nessesary, and seems like it could
+//                 be benificial to filter regions by those that are known to have CoreOS images in the
+//                 case that not every region has a CoreOS, or the correct CoreOS images.
+//
+//                 I believe that the CoreOS filtering can happen on regions, and do not believe
+//                 that CoreOS images would vary between zones within a region, but this should
+//                 be verified, because it would change the logic here and in knownZones
+
+// knownRegions is a list of AWS regions that the installer recognizes.
+// This is subset of AWS regions and the regions where RHEL CoreOS images are published.
+// The result is a map of region identifier to region description
+func knownRegions() map[string]string {
+	required := sets.NewString(rhcos.AMIRegions...)
```

Comment:
so when the rhcos pipeline is in place, this will actually come from the regions that the pipeline pushes the images to. Now, i know that for IBMCloud there is no "replicate image to all regions" like AWS. so we will only push to a single region. But it looks like the image will be mirrored to other regions offline. This will need to be discussed to see what regions to provide

Reply:
@Prashanth684 -- it does sound like we need some followup. I was under the impression that the rhcos pipeline would push to all the regions. @bpradipt did I misunderstand, and there is an offline mirroring?

Reply:
@Prashanth684 as you already mentioned there is the "replicate image to all regions" functionality like AWS. So either rhcos pipeline pushes to all regions (which might not practically scale with new regions getting added) or we handle mirroring on our own (probably via a PROW job).

We did discuss about offline mirroring but that was in the context of PowerVS default catalog. We didn't make much headway because of it being a completely backend/offline activity and the alternate approach of having the images in COS and directly importing it during install, being more flexible.

Reply:
ah...i just added a bunch of comments to the docs before seeing this. yes - if the number of regions is small , we can even consider doing this short term in the pipeline. but ideally it would be good to have a replicate command as regions expand. offline is fine too...like some sort of a cronjob that syncs the images.